prusti-interface/src/environment/mod.rs

```diff
@@ -495,10 +495,16 @@ impl<'tcx> Environment<'tcx> {
 
     /// Checks whether the given type implements the trait with the given DefId.
     pub fn type_implements_trait(&self, ty: ty::Ty<'tcx>, trait_def_id: DefId, param_env: ty::ParamEnv<'tcx>) -> bool {
```

Comment:
This method should probably not exist: I assume the empty `ty::List` will cause a crash when checking against a generic trait.

Reply:
Afaik this is not used in the `specifications` module or is used anyhow involved in constraint resolving (`constraints.rs`)?